### PR TITLE
Remove unnecessary namespacing

### PR DESCRIPTION
**Pull request by shioyama**
Great gem!!!

I noticed while reading the code that "Keisan" and other parent module names are sprinkled across the code when they aren't really necessary, since in most cases they are implied by the context. I tried removing them and specs still pass, so... I think this is ok.

Actually, I think there's more you can probably remove :smile: also I may have gone too far in a spot or two, you should double-check the code is still using the correct classes/modules. But tests pass so I guess everything is ok!

### CODE REVIEW

**Comment by shioyama** [comment]
Wow, I learned something from this little exercise: `is_a?` and `when` also prioritize the local namespace context, e.g.:

```ruby
module A
  class B
    def what_is_this?(x)
      if x.is_a?(B)
        puts "it's a B"
      else
        puts "it's not a B"
      end
    end
  end
end

class B; end

A::B.new.what_is_this?(A::B.new)
#=> "it's a B"
A::B.new.what_is_this?(B.new)
#=> "it's not a B"
```

File: lib/keisan/ast/plus.rb [hunk 1]
```diff
@@ -39,25 +39,25 @@ def simplify(context = nil)
...
             new_children + cur_child.children
           else
             new_children << cur_child
           end
         end
 
-        if children.all? {|child| child.is_a?(Keisan::AST::String)}
-          return Keisan::AST::String.new(children.inject("") do |result, child|
+        if children.all? {|child| child.is_a?(String)}
+          return String.new(children.inject("") do |result, child|
```
Comment (project-eutopia) [inline lib/keisan/ast/plus.rb]: I should have realized that namespacing rules means that `String` here will identify with `Keisan::AST::String` before the built in `String`.  Nice simplification here.

**Comment by project-eutopia** [comment]
@shioyama Thanks for your nice comments and good contribution!  I think it will improve readability to leave out these namespaces where it is clear like you have done here.

Unfortunately I just merged #47 (support for strings with multiple lines) before I saw your pull request, which introduced some conflicts due to my refactoring of `Builder` into `Builder` and `LineBuilder`.  If you rebase I will merge this PR.

Thanks again! :+1:

**Comment by shioyama** [comment]
Rebased!

**Comment by project-eutopia** [comment]
@shioyama Thanks for your contribution :+1:

**Pull request merged**